### PR TITLE
fix(connector): [Stripe] emit Stripe-Account on UCS repeat-payment flow (#16727)

### PR DESCRIPTION
**Pull request by 10X-GRACE**
## What

Thread `stripe_split_payment` through the **repeat-payment (MIT recurring charge)** flow so the UCS path emits the `Stripe-Account` header for Stripe Connect *direct* charges, matching the Direct gateway.

Sibling of #16718 (authorize/customer-create/tokenize), #16719 (capture) and #16724 (void). Resolves shadow-validation diff **#16727** (`stripe` / `repeat_payment`, merchant `yucca`/Baskhealth).

## Root cause

The `Stripe-Account` header is derived from `split_payments.stripe_split_payment.transfer_account_id` for direct charges. On the UCS repeat flow:
- the proto message `RecurringPaymentServiceChargeRequest` had **no `stripe_split_payment` field**, so hyperswitch could not send the connected-account id, and
- the proto→domain conversion (`RecurringPaymentServiceChargeRequest` → `RepeatPaymentData`) hard-coded `split_payments: None`.

The Stripe repeat `get_headers` already reads `request.split_payments` and emits the header — it just always received `None`. So `Stripe-Account` was present on Direct but absent on UCS.

## Changes

- **proto**: add `stripe_split_payment` (field `34`) to `RecurringPaymentServiceChargeRequest`
- **types**: map `stripe_split_payment` → `split_payments` in the repeat `ForeignTryFrom` via the existing `stripe_split_payment_to_domain` helper

No connector-transformer change needed — `RepeatPaymentData.split_payments` and the Stripe repeat `get_headers` already exist.

## Diff signature (before)

```
header.keyDiff.stripe-account={"hyperswitch":true,"ucs":false}
```

## Proof (after) — local shadow repro, `RecurringPaymentService.Charge`

UCS golden-log request body now carries the split payment:
```
"stripe_split_payment":{"charge_type":"direct","transfer_account_id":"acct_..."}
```

Validation service `diff_result_repeat_payment:stripe:019ed41b-fb9b-7e31-b4c3-ffd153b2062b`:
```
bodyComparison.keyDiff   = {}
headerComparison.keyDiff = {"x-merchant-id": {...}}   # benign, ignore-listed
```
`stripe-account` is gone from the diff. Both Direct and UCS send the identical `Stripe-Account` header.

> Proto field `34` matches the companion proto-rev branch `ucs-diff/stripe/16727-repeat-proto` that hyperswitch rev-pins.

### CODE REVIEW

File: crates/types-traits/domain_types/src/types.rs [hunk 1]
```diff
@@ -11844,7 +11844,7 @@ impl<
                 .map(|m| ForeignTryFrom::foreign_try_from((m, "feature data")))
```
Comment (XyneSpaces) [inline crates/types-traits/domain_types/src/types.rs:11844]: :bulb: Verify `stripe_split_payment_to_domain` helper exists in scope and handles `None` input. This path is exercised for every repeat-payment request; a missing import or panic here would break the flow.

Confirm the helper signature and add a unit test for the `stripe_split_payment: None` case.

File: crates/types-traits/grpc-api-types/proto/payment.proto [hunk 2]
```diff
@@ -3068,6 +3068,9 @@ message RecurringPaymentServiceChargeRequest {
   // Additional payment data for various payment methods
```
Comment (XyneSpaces) [inline crates/types-traits/grpc-api-types/proto/payment.proto:3068]: :warning: Adding field 34 is wire-compatible, but confirm no other concurrent PR claims this field number in `RecurringPaymentServiceChargeRequest`. Search for other open PRs touching this message.

Also regenerate the Rust proto bindings in `grpc-api-types/src/` — the diff shows only JavaScript SDK updates.

**Review by XyneSpaces — COMMENTED** [review]
## Review Summary

**Classification:** Proto-API Contract + Core Flow Framework (Stripe Connect repeat-payment splits)
**Risk Level:** Medium (proto contract change with workspace-wide impact)
**Metadata:** grace-generated-pr (author: 10X-GRACE)

---

### :warning: Warning — Proto Field Number Reservation

**Location:** `crates/types-traits/grpc-api-types/proto/payment.proto:3068-3070`

Adding field `34` to `RecurringPaymentServiceChargeRequest` is safe for wire compatibility (new optional field), but ensure no other concurrent PR is also claiming field 34 in this message. Proto field number collisions are runtime-breaking.

**Fix:** Confirm via search that no other open PRs touch `RecurringPaymentServiceChargeRequest` field numbers. Document the field reservation in any tracking issue for this diff series.



---

### :bulb: Suggestion — Verify stripe_split_payment_to_domain Helper Exists

**Location:** `crates/types-traits/domain_types/src/types.rs:11844`

The change introduces:
```rust
split_payments: stripe_split_payment_to_domain(value.stripe_split_payment),
```

**Fix:** Confirm `stripe_split_payment_to_domain` is defined and imported in this module. The helper should handle `None` input (from the proto optional) by returning `None` in the domain type. If this helper is new in this PR, verify it has unit tests.



---

### :bulb: Suggestion — Generated Rust Types Need Regeneration

**Location:** `crates/types-traits/grpc-api-types/src/`

The proto change adds field 34, but the diff only shows the JavaScript SDK generated file updated. The Rust-generated types (`*.rs` files under `grpc-api-types/src/`) must also be regenerated to include the new `stripe_split_payment` field.

**Fix:** Run the proto generation step (`make proto` or equivalent) and commit the updated Rust generated files. Without this, the Rust codebase will not compile against the new proto definition.



---

### :mag: Nitpick — Comment Accuracy

**Location:** `crates/types-traits/grpc-api-types/proto/payment.proto:3069-3070`

The comment says field 34 adds `stripe_split_payment`, but verify this aligns with the `StripeSplitPaymentRequest` message already defined elsewhere in the proto file. Cross-reference the type definition to ensure the comment and type match.

**Fix:** Add a brief doc reference: `// See: StripeSplitPaymentRequest in types.proto` (or equivalent) to aid readers navigating the codebase.



---

**Verdict:** :arrows_counterclockwise: **Request Changes** — Regenerate Rust proto bindings and verify helper exists before merge.

**Review by XyneSpaces — COMMENTED** [review]
## Review Summary

**Verdict:** :white_check_mark: Approve

**Classification:** Proto API contract + Core flow framework  
**Risk:** Low

The PR correctly adds `stripe_split_payment` field to `RecurringPaymentServiceChargeRequest` and wires it through the proto→domain conversion.

---

### Findings

No blocking issues identified. The change:
- Uses field number 34 (new, no conflict with existing fields)
- Includes descriptive proto comment explaining the field purpose
- Properly converts proto `stripe_split_payment` to domain `split_payments` via existing helper
- Generated SDK files updated consistently

### Verification Completed

- [x] Proto field number 34 is new (no reserved tag conflict)
- [x] Field comment documents purpose and header emission behavior
- [x] Domain conversion uses `stripe_split_payment_to_domain()` helper
- [x] TypeScript SDK generation updated for affected message types
- [x] Companion hyperswitch PR #12806 rev-pins to this proto change
